pkg/wekafs/apiclient/quota.go

```diff
@@ -323,6 +324,35 @@ func (a *ApiClient) GetQuotaByFileSystemAndInode(ctx context.Context, fs *FileSy
 	return ret, nil
 }
 
+// GetAllQuotasForFilesystem returns all quotas for a given filesystem
+func (a *ApiClient) GetAllQuotasForFilesystem(ctx context.Context, fs *FileSystem) ([]Quota, error) {
+	op := "GetAllQuotasForFilesystem"
+	ctx, span := otel.Tracer(TracerName).Start(ctx, op)
+	defer span.End()
+	ctx = log.With().
+		Str("trace_id", span.SpanContext().TraceID().String()).
+		Str("span_id", span.SpanContext().SpanID().String()).
+		Str("op", op).
+		Logger().WithContext(ctx)
+
+	logger := log.Ctx(ctx).With().Str("filesystem", fs.Name).Logger()
+
+	if fs == nil || fs.Uid == uuid.Nil {
+		return nil, RequestMissingParams
+	}
+
+	quotaBasePath := (&Quota{FilesystemUid: fs.Uid}).GetBasePath(a)
+	var quotas []Quota
+	err := a.Get(ctx, quotaBasePath, nil, &quotas)
+	if err != nil {
+		logger.Error().Err(err).Msg("Failed to fetch quotas for filesystem")
+		return nil, err
+	}
+
+	logger.Trace().Int("quota_count", len(quotas)).Msg("Fetched quotas for filesystem")
+	return quotas, nil
```

Comment:
This will not work without pagination. There are going to be more than 100 quotas per filesystem
Pagination support was implemented in dev branch